accepted/targeting-packs/targeting-packs.md

```diff
@@ -0,0 +1,149 @@
+  only available for Windows which makes the targeting packs indirectly Windows-
+  only too.
+
+* **It's not CI friendly**. Requiring a machine-wide install isn't friendly for
+  cloud-hosted CI machines as developers typically lack permissions to install
+  software.
+
+This document proposes to offer the targeting packs via NuGet that they can be
+acquired in a non-impactful way. It builds on the following existing specs:
+
+* [Phillip Carter](https://github.com/cartermp): [Add Target Packs on NuGet](https://github.com/dotnet/core/pull/64)
+* [Daniel Plaisted](https://github.com/dsplaisted): [Reference Assembly NuGet packages](https://gist.github.com/dsplaisted/83d67bbcff9ec1d0aff1bea1bf4ad79a)
+* [Prototype](https://github.com/dsplaisted/ReferenceAssemblyPackages)
+
+## Scenarios and User Experience
+
+* Build a project which is multi-targeted to .NET Core or .NET Standard as well
+  as .NET Framework on Mac OS or Linux using the .NET CLI
+
+* Build a project targeting Mono (which uses the same Target Framework
+  Identifier as .NET Framework) using the .NET CLI
+
+## Requirements
+
+### Goals
+
+* Doesn't require running an installer
+* Doesn't require Windows
+* Can be used in SDK-style multi-targeting projects
+
+### Non-Goals
+
+* Supporting NuGet-based acquisition of the .NET Framework runtime
+* Supporting NuGet-based acquisition from non-SDK-style projects
+* Support pre-.NET Framework 4.5 targeting packs
+* Support non-.NET Framework targeting packs (Xamarin, Mono, Unity)
+
+## Design
+
+* Daniel has created a [prototype](https://github.com/dsplaisted/ReferenceAssemblyPackages).
+    - We have an identity package
+    - Each version has a different identity and is looped in by TFM
+    - Ensures only incremental download
+    - Open question is whether we need the aggregate package at all or just
+      compute the package ID from the TFM, but that's not what we did for
+      `Microsoft.NetCore.App` or `NETStandard.Library`. What was the reason?
+
+* Includes .targets files
+    - Locate the files.
+    - Design time?
+
+### Details
+
+* There will be separate NuGet packages with reference assemblies for each
+  version of .NET Framework. This means that projects targeting a single version
+  of .NET Framework don't need to download and spend disk space on the reference
+  assemblies and intellisense files for all the other possible versions of .NET
+  Framework. It also means that when a new version of .NET Framework is
+  released, the reference assemblies for the previous versions don't need to be
+  re-shipped in an updated package along with the new assemblies.
+* There will be a single "metapackage" that can be referenced, that will have
+  conditional dependencies on each of the version-specific reference assembly
+  packages.
+* The metapackage will automatically be referenced by the .NET SDK when required
+  (ie on non-Windows OS's). On Windows, we will probably make the .NET SDK
```

Comment:
Any particular reason why the difference on Windows? It would seem simpler overall to just make it consistent...

Reply:
Agreed, not needing to install targeting pack .msi's is important for Windows based CI systems too.

Reply:
I'd be happy either way, but would lean towards requiring opt-in for this on Windows, to stay consistent with existing behavior on Windows.

Reply:
@daplaisted, what behavior difference would there be? The assemblies on the machine and those in the NuGet package should be identical.

Reply:
@tannergooding If the reference assemblies are already installed on the machine, then switching to packages means an additional download and additional disk space used.  The packages are about 20MB for each version of the framework.

I don't have a strong opinion here though.

Reply:
I'd be fine with preferring the installed targeting pack and fallback to the nuget package as long as we make it automatic.

Reply:
I agree with @dsplaisted that we should prefer the TP if its there, mostly for perf but also for compat reasons (who knows, people do all sorts of crazy things in their builds and all we can do is break them). For cases where the TP isn't installed (either non-Windows or missing TP) I like the idea of automatically using the NuGet approach. Requiring opt-in to make something build that wouldn't have build before seems silly to me.

I'll update the design section.

Reply:
It would be great if we could use the machine-wide targeting pack if it's installed, and if not fall back to using the NuGet packages.  Unfortunately I'm not sure whether that's technically feasible.

Reply:
>  Unfortunately I'm not sure whether that's technically feasible.

Oh, because you have to go through RAR to resolve it to the disk location? Hmm.... that would be sad.

Reply:
It doesn't go through RAR, but it does go through some other tasks to locate the reference assemblies, which we may not be able to run in order to determine whether to add the implicit package reference.

Reply:
Looks like we changed our mind: https://github.com/dotnet/designs/pull/33#discussion_r174899164